### PR TITLE
Gas estimation

### PR DESCRIPTION
**Pull request by Mi-Lan**
Implementing gas estimation on mainnet where possible

Currently:
1) 1Inch
2) 0xTrade
3) Uniswap

TO DO:

- UniswapV2 dexes
- Curve
- Cow

### CODE REVIEW

File: src/entities/trades/0x/0xTrade.ts [hunk 1]
```diff
@@ -152,6 +155,7 @@ export class ZeroXTrade extends TradeWithSwapTransaction {
         callData: json.data,
         value: json.value,
         priceImpact: decodeStringToPercent(json.estimatedPriceImpact, true),
+        estimatedGas: BigNumber.from(+json.estimatedGas),
```
Comment (berteotti) [inline src/entities/trades/0x/0xTrade.ts:158]: Why is there a `+` here?

File: src/entities/trades/uniswap-v2/UniswapV2.ts [hunk 2]
```diff
@@ -29,6 +28,7 @@ import {
 import { computePriceImpact, inputOutputComparator, toHex, ZERO_HEX } from './utilts'
 
 const DFYN_GAS_LIMIT = 166004
+// const GAS_ESTIMATION_LIMIT = 1000000000000000
```
Comment (berteotti) [inline src/entities/trades/uniswap-v2/UniswapV2.ts:31]: Should you delete this comment?

**Review by berteotti — COMMENTED** [review]
Apart from 2 minor things, PR looks good to me.

**Review by berteotti — APPROVED** [review]
Previous review was meant to approve.

File: src/entities/trades/0x/0xTrade.ts (hunk 1 above)
Comment (Mi-Lan) [reply src/entities/trades/0x/0xTrade.ts:158]: ohh its a typo! Will remove that

**Pull request merged**